src/debuginfo.cpp

```diff
@@ -343,7 +343,9 @@ class JuliaJITEventListener: public JITEventListener
         for (const object::SymbolRef &sym_iter : debugObj.symbols()) {
             StringRef sName;
 #ifdef LLVM37
-            sName = sym_iter.getName().get();
+            auto sNameOrError = sym_iter.getName();
+            assert(sNameOrError);
```

Comment:
In `src/Makefile` the build rule for .c files uses the variable `DISABLE_ASSERTIONS` to pass
`-DNDEBUG` to the compiler by default, so indeed assertions will be enabled by default for .c files:
```
$(BUILDDIR)/%.o: $(SRCDIR)/%.c $(HEADERS) | $(BUILDDIR)
	@$(call PRINT_CC, $(CC) $(CPPFLAGS) $(CFLAGS) $(SHIPFLAGS) $(DISABLE_ASSERTIONS) -c $< -o $@)
```

However, the build rule for .cpp files does not use `DISABLE_ASSERTIONS`:

```
$(BUILDDIR)/%.o: $(SRCDIR)/%.cpp $(SRCDIR)/llvm-version.h $(HEADERS) $(LLVM_CONFIG_ABSOLUTE) | $(BUILDDIR)
	@$(call PRINT_CC, $(CXX) $(shell $(LLVM_CONFIG_HOST) --cxxflags) $(CPPFLAGS) $(CXXFLAGS) $(SHIPFLAGS) -c $< -o $@)
```
Hence, it seems that assertions will always be enabled when compiling the `src/*.cpp` files (unless `-DNDEBUG` will be passed manually). I'm wondering if this is intended?